src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] list;
+    private int size;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        list[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        grow();
+        if (index > size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index!");
+        } else if (index == size) {
+            add(value);
+        } else {
+            System.arraycopy(list, index, list, index + 1, size - index);
+            list[index] = value;
+            size++;
+        }
+    }
 
+    private void grow() {
```

Comment:
[Use informative names for your variables and methods.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-informative-names-for-your-variables-and-methods)